File: ios/Synapse/Models/Analytics.swift
import Foundation

struct AnalyticsOverview: Codable, Sendable, Hashable {
    let totalItems: Int
    let activeItems: Int
    let totalResponses: Int
    let overallAccuracy: Double?
    let totalUsers: Int
    let completedSessions: Int
}

struct ItemStat: Codable, Identifiable, Sendable, Hashable {
    let itemId: String
    let preview: String
    let examType: String?
    let examPart: String?
    let irtB: Double?
    let calibrationStatus: String
    let responseCount: Int
    let accuracy: Double?

    var id: String { itemId }
}

struct ItemStatsResponse: Codable, Sendable, Hashable {
    let items: [ItemStat]
}

File: frontend/lib/types/analytics.ts
export interface AnalyticsOverview {
  total_items: number;
  active_items: number;
  total_responses: number;
  overall_accuracy: number | null;
  total_users: number;
  completed_sessions: number;
}

export interface ItemStat {
  item_id: string;
  preview: string;
  exam_type: string | null;
  exam_part: string | null;
  irt_b: number | null;
  calibration_status: string;
  response_count: number;
  accuracy: number | null;
}

export interface ItemStatsResponse {
  items: ItemStat[];
}
